R/lancelot.R

```diff
@@ -1960,6 +1933,88 @@ lancelot_parameters_vaccination <- function(N_tot,
   ret
 }
 
+
+lancelot_parameters_vacc_skip <- function(vacc_skip_to,
+                                          vacc_skip_progression_rate,
+                                          vacc_skip_weight,
+                                          n_vacc_classes,
+                                          n_doses,
+                                          index_dose_inverse) {
+
+  n_vacc_classes <- n_vacc_classes
+  vacc_classes <- seq_len(n_vacc_classes)
+  if (length(vacc_skip_to) != n_vacc_classes) {
+    stop(sprintf("There are %s vaccine classes so 'vacc_skip_to' must be of
+                 length %s", n_vacc_classes, n_vacc_classes))
+  }
+  if (length(vacc_skip_weight) != n_vacc_classes) {
+    stop(sprintf("There are %s vaccine classes so 'vacc_skip_weight' must be of
+                 length %s", n_vacc_classes, n_vacc_classes))
+  }
+  if (length(vacc_skip_progression_rate) != n_vacc_classes) {
+    stop(sprintf("There are %s vaccine classes so 'vacc_skip_progression_rate'
+                 must be of length %s", n_vacc_classes, n_vacc_classes))
+  }
+  if (!all(vacc_skip_to %in% vacc_classes | vacc_skip_to == 0)) {
+    stop(sprintf("There are %s vaccine classes so the values in 'vacc_skip_to'
+                 must be 0 or one of: %s",
+                 n_vacc_classes,
+                 paste(vacc_classes, collapse = ", ")))
+  }
+  if (any(vacc_skip_to == 0 & vacc_skip_weight != 0)) {
+    stop("Require 0 values in 'vacc_skip_weight' for vaccine strata that have 0
+         values in 'vacc_skip_to'")
+  }
+  if (any(vacc_skip_to == 0 & vacc_skip_progression_rate != 0)) {
+    stop("Require 0 values in 'vacc_skip_progression_rate' for vaccine strata
+         that have 0 values in 'vacc_skip_to'")
+  }
+  if (any(vacc_skip_to != 0 & vacc_skip_to <= vacc_classes + 1)) {
+    stop("Require vacc_skip_to[j] = 0 or vacc_skip_to[j] > j + 1")
+  }
+  vacc_skip_moves <- which(vacc_skip_to > 0)
+  vacc_skip_to_moves <- vacc_skip_to[vacc_skip_moves]
+  if (length(unique(vacc_skip_to_moves)) != length(vacc_skip_to_moves)) {
+    stop("Cannot have more than one vaccine skip move to the same stratum")
```

Comment:
I think this is what you meant by no duplication in the documentation; I think it'd be clearer to say no duplication in the "to" ie you can't have two skip moves from two different vaccine strata into the same one.

Reply:
I've tried to make this clearer in the parameter description